### ISSUE TITLE
[file_server browse] Layout setting is not saved

### ISSUE THREAD

**Issue by trmdi** [caddyserver/caddy#6386]
When you change the file browser's layout between Grid/List, it's reset when you leave the current page/open other directory.
It would be better if the setting remains consistent until the user changes it.

**Comment by mholt** [comment]
Agreed. Can you suggest a good way to do that that doesn't require cookies?

**Comment by trmdi** [comment]
> Agreed. Can you suggest a good way to do that that doesn't require cookies?

I don't know what is better, what's wrong with the cookie?

**Comment by mholt** [comment]
Europe's laws make the use of cookies rather unbearable.

**Comment by aliasgar55** [comment]
Maybe local or session storage?

**Comment by trmdi** [comment]
> Europe's laws make the use of cookies rather unbearable.

Caddy is already using cookies, isn't it?
https://github.com/search?q=repo%3Acaddyserver%2Fcaddy%20cookie&type=code
So I think the EU's cookie law is not an issue here.

**Comment by mholt** [comment]
The only time we _set_ cookies, AFAIK, are when the user explicitly configures load balancing policy as such. The rest of those results are related to parsing existing cookie headers, not setting.

**Comment by mholt** [comment]
> Maybe local storage?

Maybe, but that will be lost when sharing a link. You will be viewing a grid and someone else will see the list.

**Comment by trmdi** [comment]
> The only time we _set_ cookies, AFAIK, are when the user explicitly configures load balancing policy as such. The rest of those results are related to parsing existing cookie headers, not setting.

I saw there is a cookie for the Sorting button which is very similar to the Layout button.

**Comment by mholt** [comment]
Oh, interesting. I thought we got rid of those for query string parameters. In fact, I distinctly remember doing this... :thinking: We should probably remove those in favor of the query string.

**Comment by trmdi** [comment]
> Maybe local storage?

I think the cookie law is applied to similar types of cookie including localStorage.

**Comment by trmdi** [comment]
> Oh, interesting. I thought we got rid of those for query string parameters. In fact, I distinctly remember doing this... 🤔 We should probably remove those in favor of the query string.

Just curious why don't you accept the law? Maybe add an opton for caddy's admin to show/not show the cookie banner?
Because we can not avoid storing a user's preference in their browser.

**Comment by mholt** [comment]
Cookie banners are destroying the Web.

I think we can avoid it with query strings.  But it might be a little more involved since it requires changing the links.

**Comment by vtacquet** [comment]
is there a way of defaulting to grid for a certain path from a Caddyfile setting/directive?

**Comment by mholt** [comment]
Not presently, but a proposal could be considered
